### PR TITLE
fix(swap): surface THORNode errors + guard secured-asset swap destination

### PR DESCRIPTION
**Pull request by gastonm5**
## Problem

Swapping **into** a THORChain Secured Asset (e.g. `ETH-USDC`) from RUNE/BTC failed with a generic error, while swapping **out** worked.

## Root cause (API-verified) + a Step-0 code-trace correction

THORChain Secured Assets settle **on THORChain**: a swap-in mints the asset to a `thor1…` address (quote memo `=:ETH-USDC:thor1…:0/1/0`). Live mainnet testing confirmed THORNode accepts every notation the app builds (`eth-usdc-0x…`, `ETH-USDC`, `ETH-USDC-0X…`) and returns a valid quote — **but only with a valid `thor1…` destination**; a non-`thor1` destination is rejected with `code 2: "swap destination address is not the same chain as the target asset"`. So `Coin.swapAsset` is fine and needs no change.

The issue text assumed `toCoin.address` was empty / an L1 `0x…`. **The code trace shows otherwise:** a secured `toCoin` is `chain == .thorChain`, and its `.address` is derived purely from the vault key (`CoinFactory` → `CoinType.thorchain`), **independent of the secured denom** — so it already resolves to the vault's `thor1…` address (same as RUNE). The keysign memo is taken verbatim from `quote.memo` (`$0.memo = keysignPayload.memo` for THORChain-source, OP_RETURN for BTC-source), so display == signed by construction.

Given that, the thing that actually made this failure class **undiagnosable** is the error collapse: `mapThorchainSwapError` mapped every non-code-3 THORNode error to `routeUnavailable`, swallowing the real message (incl. the code-2 destination error).

## Changes

- **`SwapService.resolveThorchainDestination(toCoin:recipientAddress:)`** (new, unit-tested): preserves the existing contract (`recipientAddress ?? toCoin.address`) and, for a secured `toCoin` with no external recipient, **guards** that the resolved destination is a valid THORChain address — otherwise throws a clear, message-bearing `SwapError.serverError` instead of sending a doomed quote. Wired into `fetchCrossChainQuote` (covers both the rapid and the streaming quote, which reuse the same `destination`). Defense-in-depth on the exact failure class: a non-THORChain destination for a secured mint can never leave the device silently.
- **`THORChainHelper.isValidThorchainAddress(_:chain:)`** (new): non-throwing wrapper over the existing `validateThorchainAddress` (mainnet/chainnet/stagenet hrp).
- **`mapThorchainSwapError`**: stop collapsing non-code-3 errors to `routeUnavailable`; classify the known dust/pool/fee cases, otherwise **relay THORNode's real message** via `.serverError(message:)` (only a truly empty message falls back to `routeUnavailable`). This surfaces the code-2 destination error and any future edge case.

The from-secured, non-secured, and external-recipient paths are byte-identical.

## Tests

- `SwapThorchainDestinationTests` (new): secured ETH-USDC and a secured `btc-…` denom both resolve to the vault's `thor1…`; empty and `0x…` destinations throw; native RUNE, an EVM L1 token, and an external recipient are returned unchanged; `isValidThorchainAddress` validity check.
- `SwapErrorMappingTests` (updated): a non-code-3 error and the code-2 destination error now surface their real message; empty message still falls back to `routeUnavailable`. Existing halt / dust / pool mappings unchanged.

## Verification

- Local gate: `xcodebuild test` on `VultisigApp` (iOS Simulator, iPhone 16), worktree-local `-derivedDataPath .build-dd`. Build succeeded; **2195 tests**, the only failure being the pre-existing `"12,3%"` locale test (`StakingValidatorProjectionTests`, unrelated to this change). The three affected swap suites (`SwapThorchainDestinationTests`, `SwapErrorMappingTests`, `SwapServiceTradingHaltedTests`) re-ran green (`** TEST SUCCEEDED **`). SwiftLint clean on changed files.

## ⚠️ Needs on-device validation (manual acceptance) — THORChain swap/signing-adjacent

Please validate on a real device against live THORChain:
- RUNE → ETH-USDC and BTC → ETH-USDC complete (displayed target == signed memo target `=:ETH-USDC:thor1…`).
- From-ETH-USDC still works.
- If a swap-in still fails, the new error surfacing will now show THORNode's real message — capture it to confirm the true on-device failure mode.

Refs #4321 — not auto-closing: given the reframe (the destination is already valid in `main`, so this is defense-in-depth + diagnosability rather than a confirmed behavioral fix), this needs **on-device confirmation of the actual failure mode** (and that the swap now completes / reports a real error) before the issue is closed.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved THORChain swap destination handling for secured assets by validating/deriving the correct vault destination and blocking invalid THORChain-formatted addresses.
  * Enhanced swap error mapping so non-code-3 failures can display the original upstream error message when available, while still using clear fallbacks when messages are empty.
  * Added localized messaging for invalid secured-asset swap destinations.
* **Tests**
  * Expanded coverage for THORChain destination resolution and swap error mapping behavior.

### CODE REVIEW

**Comment by gastonm5** [comment]
## ✅ On-device acceptance — RUNE → ETH-USDC secured swap-in verified on mainnet

Live THORChain tx: [`058090F1…547D7B3C`](https://runescan.io/tx/058090F1947C473AEBB5A88FD512ED5AFDC92937D88FC37D02921729547D7B3C) — an app-originated (affiliate `vi`) swap **into** the ETH-USDC Secured Asset, **completed successfully**.

Verified via thornode (`/thorchain/tx/status`) + Midgard:

| Field | Value |
|---|---|
| Memo | `=:ETH-USDC:thor1pe0pspu4ep85gxr5h9l6k49g024vemtr80hg4c::vi:0` |
| In | `5 THOR.RUNE` from `thor1pe0psp…80hg4c` |
| Out | `1.9490 ETH-USDC-0XA0B86991…3606EB48` (secured) → `thor1pe0psp…80hg4c` |
| Destination | the vault's **own** `thor1…` address (as required for a secured mint) |
| Status | `success`, `swap_finalised: true` |

This confirms the swap-into-secured-asset path works end-to-end from the app: the destination resolves to the vault's THORChain address and the node-returned memo (`=:ETH-USDC:thor1…`) settles the secured asset to that address — matching the code trace showing `main` already sends the correct destination.

Note on scope: this is the **success path**, so it does not exercise this PR's new error-surfacing (that only appears on a failing quote). The PR's contribution remains (1) surfacing the real THORNode message instead of a generic `routeUnavailable`, and (2) the defense-in-depth guard that a secured destination is a valid THORChain address. Still worth a quick check for full coverage: **BTC → ETH-USDC** and **ETH-USDC → RUNE** (from-secured, unchanged).

File: VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift [hunk 1]
```diff
@@ -641,6 +647,42 @@ private extension SwapService {
...
+    static func resolveThorchainDestination(toCoin: Coin, recipientAddress: String?) throws -> String {
+        let destination = recipientAddress ?? toCoin.address
+
+        if recipientAddress == nil,
+           THORChainHelper.isSecuredAsset(coin: toCoin),
+           !THORChainHelper.isValidThorchainAddress(destination, chain: toCoin.chain) {
+            throw SwapError.serverError(
+                message: "Secured-asset swap destination is not a THORChain address "
+                    + "(\(destination.nilIfEmpty ?? "empty")); expected the vault's THORChain address."
+            )
```
[the remark this answers is not in the document]
Comment (gastonm5) [reply VultisigApp/VultisigApp/Blockchain/Swaps/Common/SwapService.swift:682]: Addressed in 0c31e6edf. The guard error is now localized — new key `swapSecuredAssetInvalidDestination` added to all 8 `Localizable.strings` files (en, de, es, hr, it, ko, pt, zh-Hans) and sorted via `sort_localizable.py` — and network-aware: the expected prefix is derived from `toCoin.chain` (`thor1…` mainnet / `cthor1…` chainnet / `sthor1…` stagenet) via `THORChainHelper.expectedAddressPrefix(for:)` and inserted as a format arg, alongside the offending destination value. Guard control flow and the error-surfacing logic are unchanged. Gate: full app compile (validates the strings) + the three swap suites green.

**Review by johnnyluo — APPROVED** [review]
LGTM
